FEM/fem_ele.h

```diff
@@ -170,18 +217,34 @@ class CElement
 	int Order;
 	size_t ele_dim; // Dimension of element
 	size_t dim; // Dimension of real dimension
+	size_t dim_grad; // Gradient dimension
```

Comment:
The dimensions of local and the global coordinates are different. For example, for a regular line element, dim_grad = ele_dim = 1, while for a inclined line element in 3 D space, dim_grad = 3.